omnimove_external_control/include/omnimove/external_control_message.hpp

```diff
@@ -0,0 +1,145 @@
+#ifndef EXTERNAL_CONTROL_MESSAGE_H
+#define EXTERNAL_CONTROL_MESSAGE_H
+#include <boost/asio.hpp>
+#include <cstdint>
+#include <memory>
+
+namespace omnimove
+{
+
+class ExternalControlMessage
+{
+private:
+  std::unique_ptr<char[]> msg_buffer_;
+  size_t msg_length_;
+
+protected:
+  const char * header_;
+  const char * message_;
+  const char * crc_;
+  ExternalControlMessage(const char * header, const char * message, const int & msg_length);
+
+public:
+  int16_t calculateCRC();
+  boost::asio::const_buffer getSerialisedData();
+};
+
+class ExternalControlData : public ExternalControlMessage
+{
+private:
+  const bool is_valid_;
+
+protected:
+  static constexpr const char * EXTERNAL_CONTROL_DATA_HEADER = "KMRUTV03";
+  static constexpr int EXTERNAL_CONTROL_DATA_LENGTH = 106;  // without the header and crc bytes
+  uint8_t actual_mode_;
+  int32_t actual_speed_x_;
+  int32_t actual_speed_y_;
+  int32_t actual_speed_w_;
+  uint8_t agv_is_moving_;
+  int32_t max_agv_speed_x_;
+  int32_t max_agv_speed_y_;
+  int32_t max_agv_speed_w_;
+  int32_t deviation_x_;
+  int32_t deviation_y_;
+  int32_t deviation_w_;
+  uint8_t dev_driving_;
+  uint8_t dev_target_reached_;
+  uint8_t scanning_fields_;
+  uint8_t driving_direction_;
+  uint8_t current_state_;
+  uint32_t dataMatrix_FrontCamera;
+  uint32_t dataMatrix_RearCamera_;
+  uint32_t dataMatrix_LeftCamera_;
+  uint32_t dataMatrix_RightCamera;
+  uint32_t status_optical_track_guiding_;
+  uint8_t warning_optical_track_guiding_;
+  uint8_t error_optical_track_guiding_;
+  int32_t track_deviation_x_;
+  int32_t track_deviation_y_;
+  int32_t track_deviation_w_;
+  uint8_t tag_number_front_camera_;
+  uint8_t tag_number_rear_camera_;
+  uint8_t tag_number_left_camera_;
+  uint8_t tag_number_right_camera_;
```

Comment:
As I see, these variables are not used, so they just make it harder to understand the code.
If these are to be used in the future, a controller would also be needed to broadcast the values to the ROS2 world, so at first I would just delete these and readd them later, if the ros2_controller is also ready

Reply:
removed the unused variables